### ISSUE TITLE
Mach-O header type creation fails for MH_PRELOAD (aka Kaitai: FileType.preload)

### ISSUE THREAD

**Issue by Plailect** [Vector35/binaryninja-api#1917]
When parsing a MH_PRELOAD Mach-O file, Binary Ninja throws the error `"Error during Mach-O header type creation!"`. The XNU header `mach-o/loader.h` specifies an MH_PRELOAD Mach-O as:

```
 * The file type MH_PRELOAD is an executable format intended for things that
 * are not executed under the kernel (proms, stand alones, kernels, etc).  The
 * format can be executed under the kernel but may demand paged it and not
 * preload it before execution.
```

Though poorly documented, the gist of an MH_PRELOAD Mach-O is in the name: it describes only the loading of a binary into memory. This means that *the Mach-O header itself does not end up anywhere in the final memory layout*, which is a case Binary Ninja's Mach-O parser does not seem to account for.

The loading of one of these mostly works fine, but one consequence of this behavior is that the beginning of the first segment (usually __TEXT) are parsed as a Mach-O header for the "header type creation" step, which fails spectacularly because the header was not actually relocated anywhere in the final memory layout and these bytes are actually supposed to be instructions.

The solution here is probably to either completely omit header type creation for MH_PRELOAD Mach-Os, or create a fake Mach-O header struct segment outside of the existing segments.

**Comment by Plailect** [comment]
For reproducing the issue, it looks like the LLVM project has an example MH_PRELOAD Mach-O they use for testing: [`macho-preload-x86_64`](https://reviews.llvm.org/F12711366)

**Comment by CouleeApps** [comment]
Fixed in 2.1.2350-dev.
